lib/util/itinerary.js

```diff
@@ -74,14 +76,26 @@ function getFirstTransitLeg (itinerary) {
  * (*) For transit itineraries, the first transit leg is used to make
  * the determination. Otherwise, the itinerary startTime is used.
  */
-export function getItineraryDefaultMonitoredDays (itinerary) {
+export function getItineraryDefaultMonitoredDays(itinerary) {
   const firstTransitLeg = getFirstTransitLeg(itinerary)
   const startMoment = firstTransitLeg
     ? moment(firstTransitLeg.serviceDate, 'YYYYMMDD')
     : moment(itinerary.startTime)
   const dayOfWeek = startMoment.day()
 
-  return (dayOfWeek === 0 || dayOfWeek === 6)
-    ? WEEKEND_DAYS
-    : WEEKDAYS
+  return dayOfWeek === 0 || dayOfWeek === 6 ? WEEKEND_DAYS : WEEKDAYS
+}
+
+export function itinerariesAreEqual(itinerary, other) {
+  return (
+    itinerary.legs.length === other.legs.length &&
+    itinerary.legs.every(
+      (leg, index) =>
+        other?.legs?.[index].mode === leg.mode &&
```

Comment:
Extract var for `other?.legs?.[index]`.

Reply:
And perhaps for the `from`s and `to`s, or extract a function to compare the coords of two places.

Reply:
Cleaned up in 7969cc8c08590f6daf18c60716df01360bcd547b